### PR TITLE
Bunch of doc fixes and use correct reduction operators

### LINKED ISSUE #176: Bug of doc example
- OS: Linux64
- How you installed TC (docker, conda, source): conda
- Python version: Python3.6.4
- CUDA/cuDNN version: CUDA8.0/cuDNN7.0.5
- Conda version (if using conda): 4.4.11

1. There are a doc typo bug of `tc.options` about `group conv`, the desc of `conv` is same to `group conv`. 
qq20180320-110155 2x

2. There are a doc typo bug of `avgpool` example, the `avgpool` is not sum op. 
qq20180320-142650 2x

3. There are a doc typo bug of `maxpool` example, the `max=` should be `max=!`, or it will got wrong answer if the image size not divided by the kernel size. 
qq20180320-165027 2x

By the way, I think the doc is obscure to understand.

### PR DESCRIPTION
**Pull request by prigoyal**
bunch of doc fixes spotted by our user 

closes #176

### CODE REVIEW

File: test_python/layers/test_avgpool_autotune.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def test_avgpool_autotune(self):
         # NOTE: take note of use of {{ }} below for handling TC with scalars
         LANG = """
         def avgpool(float(B, C, H, W) input) -> (output) {{
-            output(b, c, h, w) += input(b, c, h * {sH} + kh, w * {sW} + kw) where kh in 0:{kH}, kw in 0:{kW}
```
Comment (ftynse) [inline test_python/layers/test_avgpool_autotune.py]: I think we should have a warning for such uninitialized reductions if the target tensor is not input.

File: tensor_comprehensions/library/layers.yaml [hunk 2]
```diff
@@ -61,7 +61,7 @@
 - name: avgpool
   lang: |
     def avgpool(float(B, C, H, W) input) -> (output) {{
-      output(b, c, h, w) += input(b, c, h * {sH} + kh, w * {sW} + kw) where kh in 0:{kH}, kw in 0:{kW}
```
Comment (ftynse) [inline tensor_comprehensions/library/layers.yaml]: `+=!` here

File: tensor_comprehensions/library/layers.yaml [hunk 3]
```diff
@@ -61,7 +61,7 @@
...
+      output(b, c, h, w) += input(b, c, h * {sH} + kh, w * {sW} + kw) / ({kH} * {kW}) where kh in 0:{kH}, kw in 0:{kW}
     }}
 
 - name: maxpool
```
Comment (ftynse) [inline tensor_comprehensions/library/layers.yaml]: also change to `+=!` for `maxpool` here, for consistency

**Comment by ftynse** [comment]
Also, this PR changes some `.py` files which I wouldn't consider as "doc fixes"

**Comment by prigoyal** [comment]
Thanks @ftynse. I thought it was a good idea to make related changes in the
same PR. I will change the title to be more appropriate.

Yes, for redundancy there is a task open. Thank you.

**Comment by ftynse** [comment]
> I thought it was a good idea to make related changes in the same PR.

It is a good idea indeed!  Otherwise, I wouldn't have noticed the mismatch between the documentation and the code.

File: test_python/layers/test_avgpool_autotune.py (hunk 1 above)
Comment (prigoyal) [reply test_python/layers/test_avgpool_autotune.py]: agreed. Would you like to create an issue for it? I believe there was a discussion on this but there was no resolution IIRC.

Comment (ftynse) [reply test_python/layers/test_avgpool_autotune.py]: #190

**Pull request merged**